src/components/elements/DidSearch.tsx

```diff
@@ -0,0 +1,163 @@
+    };
+  });
+
+  return (
+    <Box mb={3} position="relative" ref={node} width="100%">
+      <Search color={colors.primary.base} size="24" style={{ position: "absolute", left: "10px", top: "12px" }} />
+      <Input
+        borderRadius={isOpen ? "4px 4px 0 0" : 1}
+        onChange={(event: any) => {
+          onChange(event.target.value);
+          setValue(event.target.value);
+          searchInternalIdentifiers(event.target.value);
+          searchExternalIdentifiers(event.target.value);
+          !isOpen && setIsOpen(true);
+        }}
+        borderColor={isOpen && colors.primary.base}
+        borderBottomColor={isOpen && colors.midGray}
+        p="16px 16px 16px 40px"
+        placeholder="Search by domain or DID"
+        required={required}
+        type="text"
+        value={value}
+        width="100%"
+      />
+      {((externalResults && externalResults.length > 0) || (internalResults && internalResults.length > 0)) && isOpen && (
+        <Box
+          bg={baseColors.white}
+          border={3}
+          borderTop={0}
+          borderRadius={"0 0 4px 4px"}
+          boxShadow={1}
+          position="absolute"
+          width="100%"
+          style={{ zIndex: 9 }}
+        >
+          <Box maxHeight="250px" style={{ overflow: "scroll" }}>
+            {internalResults && internalResults.length > 0 && (
+              <Box>
+                {internalResults.map((did: any, i: number) => {
+                  return (
+                    <DidSearchOptionDid
+                      key={i}
+                      onSelect={(did: string) => {
+                        setValue(did);
+                        setIsOpen(false);
+                      }}
+                      did={did.did}
+                      alias={did.alias}
+                    />
+                  );
+                })}
+              </Box>
+            )}
+            {externalResults && externalResults.length > 0 && (
+              <Box>
+                {externalResults.map((result: DidSearchResultTypes, i: number) => {
+                  return (
+                    <DidSearchOption
+                      key={i}
+                      onSelect={(did: string) => {
+                        setValue(did);
+                        setIsOpen(false);
+                      }}
+                      did={result.dids}
+                      domain={result.domain}
+                    />
+                  );
+                })}
+              </Box>
+            )}
+          </Box>
+          <Flex justifyContent="flex-end" p={3}>
+            <StyledLink href="" target="_blank">
+              <Launch color={colors.primary.base} mr={1} size="16px" style={{ verticalAlign: "text-bottom" }} />
+              Go to Serto Search
+            </StyledLink>
+          </Flex>
+        </Box>
+      )}
+    </Box>
+  );
+};
+
+export const DidSearch: React.FunctionComponent<DidSearchProps> = (props) => {
+  return (
+    <SertoSearchProvider>
```

Comment:
Makes sense here for now but if we use this in various places we could just put this provider in `SertoUiProvider` and then everything will have it.